corehq/messaging/scheduling/scheduling_partitioned/migrations/0009_update_custom_recipient_ids.py

```diff
@@ -0,0 +1,19 @@
+from django.db import migrations
+from corehq.messaging.scheduling.scheduling_partitioned.models import CaseTimedScheduleInstance
+from corehq.sql_db.util import get_db_aliases_for_partitioned_query
+
+
+def update_custom_recipient_ids(*args, **kwargs):
+    for db in get_db_aliases_for_partitioned_query():
+        CaseTimedScheduleInstance.objects.using(db).filter(recipient_id="CASE_OWNER_LOCATION_PARENT").update(
```

Comment:
I am not sure this is enough. These models are clearly quite confusing, and I am still wrapping my head around the relevant models a bit, but I think at a minimum, the actual conditional alert configuration is stored in a `CreateScheduleInstanceActionDefinition`, so to update the rules themselves and make sure future messages send correctly, you will need to update the `recipients` field in that model. These will be a bit trickier since recipients is a list of tuples, but should be a similar process.

The model you are updating represents that alert applied to a specific case, but is one of two types that does that. The other type is `CaseAlertScheduleInstance`, and I believe you need to update the recipient on those as well in the same way.

Reply:
Mmm...I see. Yeah I've done the naive approach above. I think I need to give this some more thought.